x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/util/SpatialCoordinateTypes.java

```diff
@@ -56,16 +56,30 @@ public long pointAsLong(double x, double y) {
             final long yi = XYEncodingUtils.encode((float) y);
             return (yi & 0xFFFFFFFFL) | xi << 32;
         }
+    },
+    UNSPECIFIED {
+        public Point longAsPoint(long encoded) {
```

Comment:
It'd be nice if we could pass a `Point` as a spare and write the results to it. Or something like that. Like we do with `BytesRef`.

Reply:
I think I could add an extra method to WellKnownBinary class to extract this information without ever creating an intermediate object.

Reply:
I don't know, I think WKB should always be treated as a geo_shape, need to understand why we are trying to cast wkb to a long, feels wrong.

Reply:
and I think this `UNSPECIFIED` should go away, it makes no sense to me

Reply:
Added direct access from the WKB, but there is still one object creation for the big-ending/little-endian control.

Reply:
sorry but -1 to hack wkb to access values directly

Reply:
big -1